src/catalog/catalog_accessor.cpp

```diff
@@ -0,0 +1,316 @@
+
+  // Return false here to avoid dereferencing a nullptr.
+  if (class_entry == nullptr) return false;
+
+  return class_entry->GetPtr()->SetTablePointer(table_ptr);
+}
+
+storage::SqlTable *CatalogAccessor::GetTable(table_oid_t table) {
+  auto db_handle = catalog_->GetDatabaseHandle();
+  auto class_handle = db_handle.GetClassTable(txn_, db_);
+
+  // pg_class expects the OID as a 'col_oid_t' so we need to strip the
+  // 'table_oid_t' we have to a basic integer and reconstruct it as a column OID
+  auto class_entry = class_handle.GetClassEntry(txn_, col_oid_t(!table));
+
+  // Return a nullptr if needed, before dereferencing the helper pointer.
+  if (class_entry == nullptr) return nullptr;
+
+  return &*(class_entry->GetPtr()->GetSqlTable());
+}
+
+std::vector<col_oid_t> CatalogAccessor::AddColumns(table_oid_t table, std::vector<ColumnDefinition> columns) {
+  // Blocked on fetching table entry by OID and database OID
+  // TODO(John): Check if this transaction has an outstanding update on the table entry.
+  // If not, create an update that increments the schema version field by one.
+
+  std::vector<col_oid_t> col_oids;
+  col_oids.reserve(columns.size());
+
+  std::vector<Schema::Column> catalog_columns;
+  // Loop through the column definitions and create Schema::Columns by assigning OIDs
+  for (auto const &colDef : columns) {
+    // Case off of VARLEN_COLUMN since there exist two different constructors
+    if (type::TypeUtil::GetTypeSize(colDef.GetType()) == VARLEN_COLUMN)
+      catalog_columns.emplace_back(colDef.GetName(), colDef.GetType(), colDef.GetMaxVarlenSize(), colDef.IsNullable(),
+                                   col_oid_t(catalog_->GetNextOid()));
+    else
+      catalog_columns.emplace_back(colDef.GetName(), colDef.GetType(), colDef.IsNullable(),
+                                   col_oid_t(catalog_->GetNextOid()));
+    col_oids.emplace_back(catalog_columns.back().GetOid());
+  }
+  // TODO(John): Implement this by inserting these into pg_attribute
+
+  TERRIER_ASSERT(true, "This function is not implemented yet");
+  return col_oids;
+}
+
+std::vector<bool> CatalogAccessor::DropColumns(table_oid_t table, std::vector<col_oid_t> columns) {
+  // Blocked on fetching table entry by OID and database OID
+  // TODO(John): Check if this transaction has an outstanding update on the table entry.
+  // If not, create an update that increments the schema version field by one.
+
+  std::vector<bool> col_successes;
+  col_successes.reserve(columns.size());
+
+  // Loop through the column OIDs and delete them from the underlying table
+  for (auto oid : columns) {
+    // TODO(John): Implement the metadata deletion by deleting the row in pg_attribute
+    col_successes.emplace_back(oid != INVALID_COLUMN_OID);
+  }
+
+  TERRIER_ASSERT(true, "This function is not implemented yet");
+  return col_successes;
+}
+
+bool CatalogAccessor::SetColumnNullable(table_oid_t table, col_oid_t column, bool nullable) {
```

Comment:
Instead of having separate methods for changing the column in different ways (e.g., Nullable, Default, ColumnType), would it better to have a single function where you give a new `ColumnDefinition` for an existing column and then the system figures out what changed?